app/src/main/java/com/conupods/OutdoorMaps/MapInitializer.java

```diff
@@ -50,23 +49,14 @@ public void initializeToggleButtons(Button sgwButton, Button loyButton) {
     }
 
     public void initializeLocationButton(Button locationButton) {
-        locationButton.setOnClickListener((View view) -> {
-            mCameraController.goToDeviceCurrentLocation();
-        });
+        locationButton.setOnClickListener((View view) -> mCameraController.goToDeviceCurrentLocation());
     }
 
-    public void initializeSearchBar(EditText searchBar) {
-        //TODO Remove to create custom current location button
-        searchBar.setOnEditorActionListener((TextView textView, int actionId, KeyEvent keyEvent) -> {
-            if (actionId == IME_ACTION_SEARCH
-                    || actionId == IME_ACTION_DONE
-                    || keyEvent.getAction() == KeyEvent.ACTION_DOWN
-                    || keyEvent.getAction() == keyEvent.KEYCODE_ENTER) {
-                //TODO Logic for searching goes here
+    public SearchView initializeSearchBar(SearchView searchBar, Context context) {
+        searchBar.setQueryHint("Where To?");
+        searchBar.setTransitionName("BeginTransition");
+        return searchBar;
```

Comment:
since you are passing a reference to an object and doing things to it, u dont really need to return it. if u have to do the casting, do it while calling the method. A method with a lot of instanceof is a red light

Reply:
done